code/obj/flock/structure/gnesisturret.dm

```diff
@@ -0,0 +1,147 @@
+//
+/// # Gnesis Turret - Shoots syringes full of coagulated gnesis at poor staffies
+//
+// A vat that slowly generates gnesis over time,
+/obj/flock_structure/gnesisturret
+	name = "spiky fluid vat"
+	desc = "A vat of bubbling teal fluid, covered in hollow spikes."
+	icon_state = "teleblocker-off"
+	flock_id = "Gnesis turret"
+	resourcecost = 300
+	health = 80
+	var/fluid_level_max = 250
+	var/fluid_gen_amt = 3
+	var/fluid_gen_type = "flockdrone_fluid"
+	var/fluid_shot_amt = 20
+	var/target = null
+	var/range = 8
+	var/spread = 1
+	var/datum/projectile/syringe/gnesis/current_projectile = null
+
+	var/powered = FALSE
+	// flockdrones can pass through this
+	passthrough = TRUE
+	usesgroups = TRUE
+	var/fluid_gen_cost = 30 //generating gnesis consumes compute
+	var/base_compute = 20
```

Comment:
If this is going to take compute then it needs to be really obvious that that's why it's not working, maybe add a message to the examine like "offline due to insufficient compute"